sync/awsdatabricks.py

```diff
@@ -412,6 +443,11 @@ def _monitor_cluster(
                     "utf-8",
                 )
             )
+
+            if kill_on_termination:
+                cluster_state = get_default_client().get_cluster(cluster_id).get("state")
+                if cluster_state == "TERMINATED":
+                    while_condition = False
```

Comment:
Do we actually need the `kill_on_termination` flag at all? At a glance, it looks like if you did away with it but kept the check for TERMINATED then we're set for hosted monitoring and the existing functionality is unchanged (existing functionality being the monitoring running on the job cluster, in which case it's killed when the cluster terminates anyway).

Reply:
This is mostly a safeguard on changing behavior in existing functionality and not needing to check for the cluster state repeatedly from the init script. When we use hosted monitoring, we set kill_on_termination to `true`